### PR TITLE
feat: make inflation share weight a decimal

### PR DESCRIPTION
**Pull request by shifty11**
Most changes are in tests. Non-test changes are in:
- app/upgrades/...
- x/bundles/abci.go

## Summary by CodeRabbit

- **Refactor**
  - Changed `inflation_share_weight` field type from `uint64` to `string` across multiple components for better precision handling.

- **Documentation**
  - Updated OpenAPI documentation to reflect changes in `inflation_share_weight` field type.

- **Tests**
  - Modified tests to use new `inflation_share_weight` type and methods for accurate calculations and assertions.

### CODE REVIEW

File: x/bundles/abci.go [hunk 1]
```diff
@@ -46,7 +46,7 @@ func SplitInflation(ctx sdk.Context, k bundlesKeeper.Keeper, bk util.BankKeeper,
 	for _, pool := range pk.GetAllPools(ctx) {
 		// only include active pools
 		if err := k.AssertPoolCanRun(ctx, pool.Id); err == nil {
-			totalInflationShareWeight += pool.InflationShareWeight
+			totalInflationShareWeight += uint64(pool.InflationShareWeight.TruncateInt64())
```
Comment (troykessler) [inline x/bundles/abci.go]: The reason to migrate the pool inflation share weight to a dec was so that we can have values like for example 0.2 and 0.8 instead of 2000 and 8000, please make the variable `totalInflationShareWeight` also to a dec so the calculation is accurate with every inflation share weight value

**Review by troykessler — CHANGES_REQUESTED** [review]
Also add unit tests (or change some existing ones) where the inflation share value is an actual decimal and verify that the inflation is split correctly.

File: x/bundles/abci.go (hunk 1 above)
Comment (shifty11) [reply x/bundles/abci.go]: Done

**Comment by shifty11** [comment]
> Also add unit tests (or change some existing ones) where the inflation share value is an actual decimal and verify that the inflation is split correctly.

I changed all the tests in `keeper_suite_inflation_splitting_test.go` to only use decimals.

**Comment by troykessler** [comment]
> > Also add unit tests (or change some existing ones) where the inflation share value is an actual decimal and verify that the inflation is split correctly.
> 
> I changed all the tests in `keeper_suite_inflation_splitting_test.go` to only use decimals.

What I mean here is to add tests or change existing ones with inflation share weight values between zero and one to verify the implemention of this line I requested: https://github.com/KYVENetwork/chain/blob/rapha/make-inflation-share-weight-a-decimal/x/bundles/abci.go#L44

File: x/bundles/keeper/keeper_suite_inflation_splitting_test.go [hunk 2]
```diff
@@ -1304,35 +1308,313 @@ var _ = Describe("inflation splitting", Ordered, func() {
...
+		Expect(s.App().DelegationKeeper.GetOutstandingRewards(s.Ctx(), i.STAKER_0, i.STAKER_0)).To(BeEmpty())
+
+		fundingState, _ := s.App().FundersKeeper.GetFundingState(s.Ctx(), 0)
+
+		// assert total pool funds
+		Expect(s.App().FundersKeeper.GetTotalActiveFunding(s.Ctx(), fundingState.PoolId)).To(BeZero())
+		Expect(fundingState.ActiveFunderAddresses).To(BeEmpty())
+	})
+
+	It("Produce a valid bundle with no funders, 10% inflation splitting and 0.1 inflation-share-weight", func() {
```
Comment (mbreithecker) [inline x/bundles/keeper/keeper_suite_inflation_splitting_test.go]: One would need multiple pools to test the inflation share-weight

File: x/bundles/keeper/keeper_suite_inflation_splitting_test.go [hunk 3]
```diff
@@ -1304,35 +1308,313 @@ var _ = Describe("inflation splitting", Ordered, func() {
...
+	})
+
+	It("Produce a valid bundle with no funders, 100% inflation splitting and 1 inflation-share-weight", func() {
+		// ARRANGE
+		params := pooltypes.DefaultParams()
+		params.ProtocolInflationShare = math.LegacyMustNewDecFromStr("0.1")
+		params.PoolInflationPayoutRate = math.LegacyMustNewDecFromStr("0.2")
+		s.App().PoolKeeper.SetParams(s.Ctx(), params)
+
+		pool, _ := s.App().PoolKeeper.GetPool(s.Ctx(), 0)
```
Comment (mbreithecker) [inline x/bundles/keeper/keeper_suite_inflation_splitting_test.go:1601]: Add second pool to test value of inflation-share-weight

**Review by mbreithecker — DISMISSED** [review]

**Review by mbreithecker — APPROVED** [review]

**Review by troykessler — APPROVED** [review]
